### ISSUE TITLE
Pairing and re-pairing issue with Xiaomi devices

### ISSUE THREAD

**Issue by lenny1972** [danielwelch/hassio-zigbee2mqtt#461]
**zigbee2mqtt add-on version (if edge, please report commit hash):**
1.16.0

**Operating environment (HassOS, Virtual Machine, Device/platform running Home Assistant):**

**Description of problem:**
paring 20 devices (xiaomi sensors) I cannot pair additional devices. Even deleting a paired device I cannot pair new devices,. The devices I'm not able to pair are Aquara Water Leak sensors even if I've already paired two devices of the same kind successfully. I've also paired - in the same zigbee network - two Xiaomi devices that has been recognised as "router" ( a plug and  a door-lock  ) but - even if correctly paired and they both work as expected - they are not able to act as router and accept pairing requests from the Aquara Water -Leak sensors. When trying to pair the Aquara Water Leak sensors no messages are shown from the Front-End and on zigbee2mqtt log. Distance of the sensors from the Coordinator (running on a CC2531 USB dongle) and from the two Xiaomi "routers" is about 1 mt, no other coordinators  are active and the sensors battery are brand new. I've also tried to unpair a Xiaomi button switch in order to reduce the number of devices on the network but with no luck, in addition to that the Xiaomi button switch - that was paired before - now is impossible to pair again... In the database.db file there are currently just 18 rows/entities: 1 Coordinator, 2 routers and 15 end-devices.

**Your entire configuration from the frontend (with sensitive fields redacted)**:
```json
`{
    "commit": "1.16.0",
    "config": {
        "advanced": {
            "adapter_concurrent": null,
            "availability_blacklist": [],
            "availability_blocklist": [],
            "availability_passlist": [],
            "availability_timeout": 0,
            "availability_whitelist": [],
            "cache_state": true,
            "cache_state_persistent": true,
            "cache_state_send_on_startup": true,
            "channel": 11,
            "elapsed": false,
            "ext_pan_id": [
                221,
                221,
                221,
                221,
                221,
                221,
                221,
                221
            ],
            "homeassistant_discovery_topic": "homeassistant",
            "homeassistant_legacy_triggers": true,
            "homeassistant_status_topic": "hass/status",
            "last_seen": "disable",
            "legacy_api": true,
            "log_directory": "/share/zigbee2mqtt/log/%TIMESTAMP%",
            "log_file": "log.txt",
            "log_level": "debug",
            "log_output": [
                "console",
                "file"
            ],
            "log_rotation": true,
            "log_syslog": {},
            "pan_id": 6754,
            "report": false,
            "soft_reset_timeout": 0,
            "timestamp_format": "YYYY-MM-DD HH:mm:ss"
        },
        "ban": [],
        "blocklist": [],
        "device_options": {},
        "devices": {
            "0x00158d000130ec12": {
                "friendly_name": "PIR n.1"
            },
            "0x00158d00015b62bd": {
                "friendly_name": "Porta Casa"
            },
            "0x00158d00015c7d98": {
                "friendly_name": "Condizionatore"
            },
            "0x00158d00016d065c": {
                "friendly_name": "Temperatura & Umidità Cameretta"
            },
            "0x00158d00016d0c08": {
                "friendly_name": "Temperatura & Umidità Bagno Principale"
            },
            "0x00158d00016d0c88": {
                "friendly_name": "Temperatura & Umidità Bagno in Camera"
            },
            "0x00158d00016d0d14": {
                "friendly_name": "Temperatura & Umidità Salone"
            },
            "0x00158d0001a2b575": {
                "friendly_name": "Temperatura & Umidità Camera da Letto"
            },
            "0x00158d0001d5944e": {
                "friendly_name": "Allagamento Bagno in Camera"
            },
            "0x00158d0001e73b3d": {
                "friendly_name": "Porta Freezer"
            },
            "0x00158d0001e73b97": {
                "friendly_name": "Porta Frigorifero"
            },
            "0x00158d000200b341": {
                "friendly_name": "Sensore Livello Sump"
            },
            "0x00158d000200b3f4": {
                "friendly_name": "Sensore Vasca di Osmosi"
            },
            "0x00158d000237266c": {
                "friendly_name": "Allagamento Terrazzo"
            },
            "0x00158d00028e2eb2": {
                "friendly_name": "Serratura Porta"
            },
            "0x00158d00057f8078": {
                "friendly_name": "PIR n.3"
            },
            "0x00158d0005818a36": {
                "friendly_name": "PIR n.2"
            }
        },
        "experimental": {
            "new_api": true,
            "output": "json"
        },
        "external_converters": [],
        "frontend": {
            "port": 8099
        },
        "groups": {},
        "homeassistant": true,
        "map_options": {
            "graphviz": {
                "colors": {
                    "fill": {
                        "coordinator": "#e04e5d",
                        "enddevice": "#fff8ce",
                        "router": "#4ea3e0"
                    },
                    "font": {
                        "coordinator": "#ffffff",
                        "enddevice": "#000000",
                        "router": "#ffffff"
                    },
                    "line": {
                        "active": "#009900",
                        "inactive": "#994444"
                    }
                }
            }
        },
        "mqtt": {
            "base_topic": "zigbee2mqtt",
            "include_device_information": false,
            "server": "mqtt://core-mosquitto",
            "user": "**********"
        },
        "passlist": [],
        "permit_join": false,
        "queue": {},
        "serial": {
            "disable_led": false,
            "port": "/dev/ttyACM0"
        },
        "whitelist": []
    },
    "coordinator": {
        "meta": {
            "maintrel": 3,
            "majorrel": 2,
            "minorrel": 6,
            "product": 0,
            "revision": 20190608,
            "transportrev": 2
        },
        "type": "zStack12"
    },
    "log_level": "debug",
    "network": {
        "channel": 11,
        "extended_pan_id": "0xdddddddddddddddd",
        "pan_id": 6754
    },
    "permit_join": true,
    "version": "1.16.0"
}`
```

**Your logs from Home Assistant**

No errors in the Homeassistant log when trying to pair the devices.

**Comment by ciotlosm** [comment]
Try adding a USB extension cord to move the usb stick away from your RPI. Also would be nice to see some information around the adapter if it has the right version. 

Typically your issue is more for the zigbee2mqtt repo as the add-on just wraps around that (https://github.com/Koenkk/zigbee2mqtt/issues/new/choose)

**Comment by lenny1972** [comment]
Marius thanks for the prompt reply. The USB CC2351 dongle is already connected to the RPi via a USB male-female cable. The firmware running on the adapter is the CC2531_DEFAULT_20190608.zip available on Github (https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_Home_1.2/bin/default). If this info will not give you any clue in order to solve the issue I'll open an issue on the original project.

**Comment by ciotlosm** [comment]
@lenny1972 Can you try running the source routing firmware?

**Comment by lenny1972** [comment]
I've already the source routing firmware installed on my CC2351 dongle. Anyway I found the reason of sensors unable to pair. Reading this article [()]https://community.hubitat.com/t/xiaomi-aqara-devices-pairing-keeping-them-connected/623/55?u=veeceeoh)] I found that my fritzbox router (broadcasting the wifi signal on both 2.4 Ghz and 5 Ghz) probably caused interference to the CC2351 dongle zigbee  network. I simply switch in the zigbee2mqtt configuration the channel from the 11 (the default one) to 25 that is reported to work good with Xiaomi devices/sensors (a lot of interesting info here [()]https://community.hubitat.com/t/xiaomi-aqara-devices-pairing-keeping-them-connected/623 and everything worked like a charm. For people having my same pairing issue should be a solution. IMPORTANT: Take in mind that changing the network channel require to pair all the sensors again..
